src/UserService.Business/Commands/Avatar/AddAvatarsCommand.cs

```diff
@@ -0,0 +1,150 @@
+
+    public AddAvatarsCommand(
+      IAvatarRepository avatarRepository,
+      IUserRepository userRepository,
+      IAccessValidator accessValidator,
+      IAddAvatarRequestValidator requestValidator,
+      IDbUserAvatarMapper dbUserAvatarMapper,
+      ICreateImageDataMapper createImageDataMapper,
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<ICreateImagesRequest> rcCreateImage,
+      ILogger<AddAvatarsCommand> logger)
+    {
+      _avatarRepository = avatarRepository;
+      _userRepository = userRepository;
+      _accessValidator = accessValidator;
+      _requestValidator = requestValidator;
+      _dbUserAvatarMapper = dbUserAvatarMapper;
+      _createImageDataMapper = createImageDataMapper;
+      _httpContextAccessor = httpContextAccessor;
+      _rcCreateImage = rcCreateImage;
+      _logger = logger;
+    }
+
+    private List<Guid> AddImages(List<AddImageRequest> request, Guid userId, Guid senderId, List<string> errors)
+    {
+      if (request == null || request.Contains(null))
+      {
+        return null;
+      }
+
+      List<Guid> result = null;
+
+      string errorMessage = "Can not add images. Please try again later.";
+      string logMessage = "Errors while adding images.";
+
+      try
+      {
+        Response<IOperationResult<ICreateImagesResponse>> createResponse = _rcCreateImage.GetResponse<IOperationResult<ICreateImagesResponse>>(
+          ICreateImagesRequest.CreateObj(
+            _createImageDataMapper.Map(request, senderId),
+            ImageSource.User))
+          .Result;
+
+        if (createResponse.Message.IsSuccess)
+        {
+          result = createResponse.Message.Body.ImagesIds;
+
+          List<DbUserAvatar> dbUsersAvatarsCreate = _dbUserAvatarMapper.Map(result, userId);
+
+          _avatarRepository.Create(dbUsersAvatarsCreate);
+        }
+        else
+        {
+          string warningMessage = logMessage + " Errors: {Errors}";
+
+          _logger.LogWarning(
+            warningMessage,
+            string.Join('\n', createResponse.Message.Errors));
+
+          errors.Add(errorMessage);
+        }
+      }
+      catch (Exception e)
+      {
+        _logger.LogError(e, logMessage);
+
+        errors.Add(errorMessage);
+      }
+
+      return result;
+    }
+
+    public OperationResultResponse<List<Guid>> Execute(AddAvatarRequest request)
+    {
+      OperationResultResponse<List<Guid>> response = new();
+      List<string> errors = new();
+
+      Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+      DbUser dbUser = _userRepository.Get(senderId);
+
+      if (!(dbUser.IsAdmin ||
+            _accessValidator.HasRights(Rights.AddEditRemoveUsers))
+            && senderId != request.UserId)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.Add("Not enough rights.");
+      }
+      else
```

Comment:
I don't think it needs 'else', maybe look at the command structure in other services